theatre/studio/src/uiComponents/form/BasicNumberInput.tsx

```diff
@@ -275,14 +270,23 @@ const BasicNumberInput: React.FC<{
     return {
       inputChange,
       onBlur,
-      transitionToDraggingMode,
       onInputKeyDown,
       onClick,
-      onFocus,
       onDragEnd,
       onDrag,
+      transitionToEditingViaKeyboardMode,
+      transitionToDraggingMode,
     }
-  }, [])
+  }, []) // TODO: fix the missing dependency warning
```

Comment:
This will need quite a bit of refactoring so to reduce the noise in this PR I'll do this separately

Reply:
In order to fix a couple of weird bugs I ended up refactoring as part of this PR, so not follow up PR now.